spec/Powertrain/Battery.vspec

```diff
@@ -290,12 +320,13 @@ Charging.ChargePlugType:
 Charging.Mode:
   datatype: string
   type: actuator
-  allowed: ['MANUAL', 'TIMER', 'GRID', 'PROFILE']
+  allowed: ['MANUAL', 'TIMER', 'GRID', 'PROFILE', 'V2X']
```

Comment:
There were a short discussion on this signal last week - if V2X as name is misleading, if we need an IDLE state, if we need to split it into charge/discharge mode signals... That made me think - how is this signal intended to be used and does the current state reflect that. I am no expert in this area so feedback is appreciated. If we are to capture "modes" from a vehicle perspective, does the current "state space" reflects that? Do we possible need other states and a table, possibly like below.

I.e. That you in the vehicle (by settings) can control what shall happen if a charging cable is connected, and if the vehicle only will accept charge or also discharge? But do we then need to separate "GRID" from other charge/discharge modes? I.e. will the vehicle behave differently if you connect a "less intelligent load" like your boat or off-grid cottage or if you connect to a "smart Grid"? Is the difference possibly that in "manual/triggered discharge" we accept any load, but in grid-mode the "load/charger" must identify itself towards the vehicle with some trusted credentials.

Mode | What happens when cable is connected? | When will charging start? | When will discharge start?
--- |   --- | --- | ---
IDLE | Nothing	 | Never	 | Never
MANUAL	 | Charging starts	 | Cable connected 	 | Never
TRIGGERED	 | User dialog appears	 | When acknowledged by user	 | Never
TIMER	 | Depends on timer	 | Depends on timer	 | Never
GRID	 | Controlled by grid if authorized	 | Controlled by grid if authorized	 | Controlled by grid if authorized	
PROFILE	 | Depends on profile	 | Depends on profile	 | Never
MANUAL_CHARGE_DISCHARGE	 | Charge or Discharge starts	 | Cable connected	 | Cable connected
TRIGGERED_CHARGE_DISCHARGE	 | User dialog appears	 | When acknowledged by user	 | When acknowledged by user

One could off course also think off other combinations, like charging may start as soon as a charge is connected, but for discharge the driver needs to acknowledge it. Then having separate signals would be a benefit, potentially with different states.

If this discussion take too long time we could possibly handle it separately from the other changes.

Reply:
nitpick: "MANUAL_CHARGE_DISCHARGE" is not clearly defined, so WHAT does happen when a cable is connected? Also generally not sure if "manual" is the correct term indicating a mode where connecting a cable AUTOMATICALLY does something

Reply:
> nitpick: "MANUAL_CHARGE_DISCHARGE" is not clearly defined, so WHAT does happen when a cable is connected? Also generally not sure if "manual" is the correct term indicating a mode where connecting a cable AUTOMATICALLY does something

AUTOMATIC_CHARGE_DISCHARGE is maybe a better term. My assumption here is that the vehicle charge port here would act similar to a USB-port on a laptop. If you connect a charger the laptop will start charging. If you connect your phone (i.e. a load) the laptop will start charging your phone. I.e. the item you connect decide if it will charge or discharge the laptop. But I have no clue if that is how it will work also for a vehicle.